stdlib/2and3/shutil.pyi

```diff
@@ -1,34 +1,51 @@
-# Stubs for shutil
 import os
 import sys
 
-# Based on http://docs.python.org/3.2/library/shutil.html
-
 # 'bytes' paths are not properly supported: they don't work with all functions,
 # sometimes they only work partially (broken exception messages), and the test
 # cases don't use them.
 
 from typing import (
     List, Iterable, Callable, Any, Tuple, Sequence, NamedTuple, IO,
-    AnyStr, Optional, Union
+    AnyStr, Optional, Union, Set, TypeVar, overload, Type
 )
 
 if sys.version_info >= (3, 6):
     _Path = Union[str, os.PathLike[str]]
+    _AnyStr = str
+    _AnyPath = TypeVar("_AnyPath", str, os.PathLike[str])
```

Comment:
Reminder to self: I'm worried about these conditionally defined typevars. Do they make sense for all use cases?

Reply:
Side note: Unfortunately, in Python 3 shutil is a bit of a mess, type-wise. As is alluded to in the stub, bytes is often does not work as a path and whether the returned path is a `str` or a `Path` is dependent on the phase of the moon. I am seriously contemplating to propose patches for the stdlib to fix this for 3.8, but of course that wouldn't help for the 3.6 and 3.7 stubs.